OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -167,12 +173,20 @@ public override bool Tick(Actor self)
 				return true;
 
 			var isSlider = aircraft.Info.CanSlide;
+			var desiredFacing = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : aircraft.Facing;
 			var move = isSlider ? aircraft.FlyStep(desiredFacing) : aircraft.FlyStep(aircraft.Facing);
 
-			// Inside the minimum range, so reverse if we CanSlide
-			if (isSlider && insideMinRange)
+			// Inside the minimum range, so reverse if we CanSlide, otherwise face away from the target.
```

Comment:
If the plane is not past the center of the target annulus and the turn radius is much smaller than the minimum range, the plane will instantly turn around. If the turn radius is much larger it will keep moving forward with a slight bend until it reaches the target annulus at the other side. If the plane is past the center, it will just keep moving forward in a straight line. There is never a problem here.

I think you are a bit confused here: This has nothing to do with `AttackTurnDelay`; This is just generic `MoveWithinRange` behaviour.

The problem you are thinking of could theoretically occur, but only when the turn radius is _bigger_ than the _maximum_ weapon range. This case is already covered by the block ate lines 225&ndash;244 below.